lib/ae_mdw/fields.ex

```diff
@@ -0,0 +1,70 @@
+defmodule AeMdw.Fields do
+  @moduledoc """
+  A simple fields querying API to filter transactions by type and/or ID.
+  """
+
+  alias AeMdw.Collection
+  alias AeMdw.Db.Model
+  alias AeMdw.Db.State
+  alias AeMdw.Db.Util, as: DbUtil
+  alias AeMdw.Node
+  alias AeMdw.Node.Db
+  alias AeMdw.Txs
+  alias AeMdw.Util
+
+  @typep state() :: State.t()
+  @typep pubkey() :: Db.pubkey()
+  @typep direction() :: Collection.direction()
+  @typep range() :: {:gen, Range.t()} | nil
+  @typep cursor() :: Txs.tx() | nil
+
+  @create_tx_types ~w(contract_create_tx channel_create_tx oracle_register_tx name_claim_tx ga_attach_tx)a
```

Comment:
Thank you for clarifying, I removed `name_claim_tx` from the list, since I think that these (together with `contract_create_tx`) should be handled differently. The account ID given in the URL does not appear in the tx object being return for these types of events. A new type of event would make more sense (next PR)

Reply:
`:contract_create_tx` may be kept for the contract account (would only add the `:contract_call_tx` for child contracts)

Reply:
The `contract_create_tx` is already included. I wanted contracts created through transaction calls to be a separate type of event so we are able to transmit to the user that this isn't a contract call made TO the contract, but to a different contract that creates the account given

Reply:
> Thank you for clarifying, I removed `name_claim_tx` from the list, since I think that these (together with `contract_create_tx`) should be handled differently. The account ID given in the URL does not appear in the tx object being return for these types of events. A new type of event would make more sense (next PR)

I was referring to "the account ID given in the URL does not appear, ... A new type of event would make more sense" to say that `:contract_create_tx` works "for the contract account".

Likewise the `:contract_create_tx`, the child contract account might appear on the contract call but I also think that it would be better leave it for another kind of event. The only request here is that it should be uniform, if a contract creation will be handled for contract account in a different event, the same would need to be done for child contracts and generalized accounts (or if kept `:contract_call_tx` would be added).

Reply:
You're right, I hadn't realized that child contracts were also included in the Field model. I've now removed these to then add them as a separate event later. Thank you!